plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/connection/Connection.java

```diff
@@ -314,12 +314,20 @@ private IAuthorizationStrategy getAuthorizationStrategy() {
 	 * @return
 	 */
 	public boolean isConnected(IProgressMonitor monitor) {
-		client.setAuthorizationStrategy(getAuthorizationStrategy());
+		boolean needStrategy = client.getAuthorizationStrategy() == null;
```

Comment:
I see the following code being repeated several times:
```
boolean needStrategy = client.getAuthorizationStrategy() == null;
if(needStrategy) {
	client.setAuthorizationStrategy(getAuthorizationStrategy());
}
```

I think that if this patch solves the issue (which I currently cant reproduce and thus dont know if it helps) we could move the above to a separate method called #updateAuthorizationStrategy() (or similar)